app/main.py

```diff
@@ -1,8 +1,35 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        self.husband = None
+        self.wife = None
+        Person.people[name] = self
+
+    def set_relationships(self, people: dict) -> dict:
+        if "wife" in people and people["wife"]:
+            self.wife = Person.people[people["wife"]]
+        elif "wife" in people:
+            if hasattr(self, "wife"):
+                del self.wife
+
+        if "husband" in people and people["husband"]:
+            self.husband = Person.people[people["husband"]]
```

Comment:
Similar to the wife's check, ensure that the husband's name exists in `Person.people` before accessing it to prevent `KeyError`.